Modules/Scripted/DICOMLib/DICOMWidgets.py

```diff
@@ -386,6 +388,92 @@ def onDatabaseDirectoryChanged(self,databaseDirectory):
     if slicer.dicomDatabase:
       slicer.app.setDICOMDatabase(slicer.dicomDatabase)
 
+  def onDirectoryImported(self):
+    """The dicom browser will emit multiple directoryImported
+    signals during the same operation, so we collapse them
+    into a single check for compatible extensions."""
+    if not self.extensionCheckPending:
+      self.extensionCheckPending = True
+      def timerCallback():
+        self.promptForExtensions()
+        self.extensionCheckPending = False
+      qt.QTimer.singleShot(0, timerCallback)
+
+  def promptForExtensions(self):
+    extensionsToOffer = self.checkForExtensions()
+    if len(extensionsToOffer) != 0:
+      if len(extensionsToOffer) == 1:
+        pluralOrNot = " is"
+      else:
+        pluralOrNot = "s are"
+      message = "The following extension%s not installed, but may help you work with DICOM data in your database:\n\n" % pluralOrNot
+      for extension in extensionsToOffer:
+        message += '  ' + extension + '\n'
+      message += "\n\nYou can install extensions using the Extension Manager option from the View menu."
+      slicer.util.infoDisplay(message, windowTitle='DICOM')
+
+  def checkForExtensions(self):
```

Comment:
-> `checkForExtensions(dicomDatabase, extensionDescriptorPath)`

The idea is to make the function independent of the internal state and potentially re-usable if different `database` and `descriptorfile`